src/zenml/models/pipeline_deployment_models.py

```diff
@@ -0,0 +1,116 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing pipeline deployments."""
+
+from typing import TYPE_CHECKING, Dict, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from zenml.config.pipeline_configurations import PipelineConfiguration
+from zenml.config.step_configurations import Step
+from zenml.models.base_models import (
+    WorkspaceScopedRequestModel,
+    WorkspaceScopedResponseModel,
+)
+from zenml.models.filter_models import WorkspaceScopedFilterModel
+
+if TYPE_CHECKING:
+    from zenml.models import (
+        PipelineBuildResponseModel,
+        PipelineResponseModel,
+        ScheduleResponseModel,
+        StackResponseModel,
+    )
+
+# ---- #
+# BASE #
+# ---- #
+
+
+class PipelineDeploymentBaseModel(BaseModel):
+    """Base model for pipeline deployments."""
+
+    run_name_template: str
+    pipeline_configuration: PipelineConfiguration
+    step_configurations: Dict[str, Step] = {}
+    client_environment: Dict[str, str] = {}
```

Comment:
This could benefit from some Field help hints to be shown in the rendered OpenAPI spec.